contracts/singletons/SystemDeployerV1.sol

```diff
@@ -0,0 +1,883 @@
+    // CONSTRUCTOR
+    // ======================================================================
+
+    constructor(address systemDeployerEventEmitter_) {
+        _systemDeployerEventEmitter = systemDeployerEventEmitter_;
+    }
+
+    // ======================================================================
+    // ISystemDeployer
+    // ======================================================================
+
+    // --- View Functions ---
+
+    function predictProxyAddress(
+        address implementation_,
+        bytes calldata initData_,
+        bytes32 salt_,
+        address deployer_
+    ) public view override returns (address) {
+        if (implementation_.code.length == 0) {
+            revert ImplementationMustBeAContract();
+        }
+
+        // Calculate the proxy bytecode (implementation address + init data)
+        bytes memory bytecode = abi.encodePacked(
+            type(ERC1967Proxy).creationCode,
+            abi.encode(implementation_, initData_)
+        );
+
+        // Calculate the CREATE2 address
+        bytes32 hash = keccak256(
+            abi.encodePacked(
+                bytes1(0xff),
+                deployer_,
+                salt_,
+                keccak256(bytecode)
+            )
+        );
+
+        return address(uint160(uint256(hash)));
+    }
+
+    function systemDeployerEventEmitter()
+        public
+        view
+        virtual
+        override
+        returns (address)
+    {
+        return _systemDeployerEventEmitter;
+    }
+
+    // --- State-Changing Functions ---
+
+    function deployProxy(
+        address implementation_,
+        bytes memory initData_,
+        bytes32 salt_
+    ) public returns (address) {
+        if (implementation_.code.length == 0) {
+            revert ImplementationMustBeAContract();
+        }
+
+        address proxy = address(
+            new ERC1967Proxy{salt: salt_}(implementation_, initData_)
+        );
+
+        emit ProxyDeployed(proxy, implementation_);
+
+        return proxy;
+    }
+
+    function setupSafe(
+        address safeProxyFactory_,
+        bytes32 salt_,
+        VotesERC20Params calldata votesERC20Params_,
+        AzoriusGovernanceParams calldata azoriusGovernanceParams_,
+        ModuleFractalV1Params calldata moduleFractalV1Params_,
+        FreezeParams calldata freezeParams_
+    ) public virtual override {
+        // create a two dimensional array to hold the new VotesERC20V1 and VotesERC20LockableV1 addresses
+        address[2][] memory allNewVotesERC20Addresses = new address[2][](2);
+
+        _deployAllVotesERC20(
+            salt_,
+            votesERC20Params_,
+            allNewVotesERC20Addresses
+        );
+
+        address azoriusModuleAddress = _deployAzoriusGovernance(
+            salt_,
+            azoriusGovernanceParams_,
+            allNewVotesERC20Addresses
+        );
+
+        _deployModuleFractal(salt_, moduleFractalV1Params_);
```

Comment:
I know we used to go by Fractal, and the contracts still reflect that. With this new PR, do you think it might be a good time to start considering a rename to Decent?

Reply:
There is _one_ contract still with the word `Fractal` in it: The `ModuleFractalV1` contract. I don't plan on renaming that, it is the module that enables a parent to have control over a child, and i think the name encompasses the purpose quite nicely.

Reply:
make sense!